base/applications/regedit/framewnd.c

```diff
@@ -807,6 +814,98 @@ static void ChooseFavorite(LPCWSTR pszFavorite)
         RegCloseKey(hKey);
 }
 
+static LPWSTR GetItemFullPath(HTREEITEM hTI)
+{
+    HKEY hRoot;
+    WCHAR rootname[MAX_PATH], *buffer;
+    SIZE_T rootlen, subkeylen;
+    LPCWSTR subkey = GetItemPath(g_pChildWnd->hTreeWnd, hTI, &hRoot);
+    if (!subkey || !hRoot)
+        return NULL;
+    if (!GetKeyName(rootname, ARRAY_SIZE(rootname), hRoot, L""))
+        return NULL;
+    rootlen = lstrlenW(rootname) + 1; // + 1 for '\\'
+    subkeylen = lstrlenW(subkey);
+    buffer = (WCHAR*)GlobalAlloc(GMEM_FIXED, (rootlen + subkeylen + 1) * sizeof(WCHAR));
+    if (buffer)
+    {
+        lstrcpyW(buffer, rootname);
+        buffer[rootlen - 1] = '\\';
+        lstrcpyW(buffer + rootlen, subkey);
+    }
+    return buffer;
+}
+
+static INT_PTR CALLBACK AddToFavoritesDlgProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    HWND hName = GetDlgItem(hWnd, IDC_FAVORITENAME);
+    WCHAR name[MAX_PATH];
+    switch (uMsg)
+    {
+        case WM_INITDIALOG:
+        {
+            TVITEM tvi;
+            tvi.mask = TVIF_HANDLE | TVIF_TEXT;
+            tvi.hItem = TreeView_GetSelection(g_pChildWnd->hTreeWnd);
+            tvi.pszText = name;
+            tvi.cchTextMax = _countof(name);
+            if (!TreeView_GetItem(g_pChildWnd->hTreeWnd, &tvi))
+                tvi.pszText[0] = UNICODE_NULL;
+            SendMessageW(hName, WM_SETTEXT, 0, (LPARAM)tvi.pszText);
+            SendMessageW(hName, EM_LIMITTEXT, _countof(name) - 1, 0);
+            return TRUE;
+        }
+        case WM_COMMAND:
+            switch (LOWORD(wParam))
+            {
+                case IDOK:
+                    {
+                        LPCWSTR path;
+                        HKEY hKey;
+                        DWORD err;
+                        if (GetWindowTextW(hName, name, _countof(name)))
+                            err = ERROR_SUCCESS;
+                        else
+                            err = GetLastError();
+
+                        if (err == ERROR_SUCCESS)
+                        {
+                            path = GetItemFullPath(NULL);
```

Comment:
Regarding the "NULL": Wouldn't it be equivalent to use `TVI_ROOT` here instead? (at least this other value name conveys self-documentation)

Reply:
No, I want the current item, not the root. See all the other calls to GetItemPath in this file.

Reply:
ah, correct. GetItemPath does
```c
 107     if (!hItem)
 108     {
 109         hItem = TreeView_GetSelection(hwndTV);
 110     }
```
so this is what you want.